File: Reports/Daniil_Pozdnyakov/lab4/src/task3/Book.cs
using System;

namespace lab4._3._5
{
    public class Book
    {
        private int Year = 0;
        private string Title, Author;

        public Book(int year, string title, string author)
        {
            Year = year;
            Title = title;
            Author = author;
        }

        public void SetYear(int year)
        {
            Year = year;
        }

        public int GetYear()
        {
            return Year;
        }

        public void SetTitle(string title)
        {
            Title = title;
        }

        public string GetTitle()
        {
            return Title;
        }

        public void SetAuthor(string author)
        {
            Author = author;
        }

        public string GetAuthor()
        {
            return Author;
        }

        public bool _Equals(object c_Other)
        {
            if (GetYear().Equals(((Book)c_Other).GetYear()) && GetTitle().Equals(((Book)c_Other).GetTitle()) && GetAuthor().Equals(((Book)c_Other).GetAuthor()))
            {
                return true;
            }
            return false;
        }
    }
}

File: reports/Valchuk/lab5/src/task1/Book.java
public abstract class Book {
    protected String cipher;
    protected String author;
    protected String title;
    protected Short year;
    protected String publisher;

    public Book() {}

    public Book(String cipher, String author, String title, Short year, String publisher) {
        this.cipher = cipher;
        this.author = author;
        this.title = title;
        this.year = year;
        this.publisher = publisher;
    }

    public String getCipher() {
        return cipher;
    }

    public String getAuthor() {
        return author;
    }

    public String getTitle() {
        return title;
    }

    public Short getYear() {
        return year;
    }

    public String getPublisher() {
        return publisher;
    }

    public void setCipher(String cipher) {
        this.cipher = cipher;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public void setYear(Short year) {
        this.year = year;
    }

    public void setPublisher(String publisher) {
        this.publisher = publisher;
    }
}
